storage/postgresql/save.go

```diff
@@ -9,30 +9,13 @@ import (
 )
 
 func (b *PostgresBackend) SaveEvent(ctx context.Context, evt *nostr.Event) error {
-	// react to different kinds of events
-	if evt.Kind == nostr.KindSetMetadata || evt.Kind == nostr.KindContactList || (10000 <= evt.Kind && evt.Kind < 20000) {
-		// delete past events from this user
-		b.DB.ExecContext(ctx, `DELETE FROM event WHERE pubkey = $1 AND kind = $2`, evt.PubKey, evt.Kind)
-	} else if evt.Kind == nostr.KindRecommendServer {
-		// delete past recommend_server events equal to this one
-		b.DB.ExecContext(ctx, `DELETE FROM event WHERE pubkey = $1 AND kind = $2 AND content = $3`,
-			evt.PubKey, evt.Kind, evt.Content)
-	} else if evt.Kind >= 30000 && evt.Kind < 40000 {
-		// NIP-33
-		d := evt.Tags.GetFirst([]string{"d"})
-		if d != nil {
-			b.DB.Exec(`DELETE FROM event WHERE pubkey = $1 AND kind = $2 AND tagvalues && ARRAY[$3]`,
-				evt.PubKey, evt.Kind, d.Value())
-		}
+	deleteQuery, deleteParams, shouldDelete := deleteBeforeSaveSql(evt)
+	if shouldDelete {
+		_, _ = b.DB.ExecContext(ctx, deleteQuery, deleteParams...)
```

Comment:
There's opportunity for error handling here but I didn't want to change any of the existing behavior without double checking.